### ISSUE TITLE
Testing trained Autoencoders

### ISSUE THREAD

**Issue by Bedinka** [Bedinka/PPI-prediction-with-Deep-Learning-Language-Model-#13]
# PPI Analysis with Deep Learning Language Model Project

## Description

With different inputs, its needed to evaluate the performance and and validate the output encoded vectors of the trained autoencoder models. 
Spearman Correalation was computed on the input and the outputs to show the correlation and accuracy. 
( dist_ca_train,encoded_vectors_train) 

### The changes to the file can be found [here](https://github.com/lionking0000/YangLabIntern/blob/master/PPI_W_DLLM/test_autoencoder.py).
---

### Problem Statement

Based on these collected data during the trainings , evaluation was run to determine the best model for encoding. 

     "model_name": model_name,
     "epochs" : epoch,
     "latent_dim" : latent_dim,
     "matrix_size" : size,
     "processed_sample" : processed_sample,
     "train_losses": loss_history.train_losses,
      "val_losses": loss_history.val_losses,
      "spearman_correlation": correlation ,
      "spearman_p_value": p_value,
      "spearman_correlation_dim1_2": correlation2,
      "spearman_p_value_dim1_2": p_value2 

---

### Proposed Solution

The collected Data was plotted in several ways to show in a humanly readable graphs for later selections.
The autoencoder_data_collet.py script is the automatized to run all the possible senerios and create plots. 

---

### Expected Behavior (for bugs)
[Explain what you expected to happen when following the steps above.]

---

### Actual Behavior (for bugs)
[Explain what actually happened when following the steps above.]

---

### Additional Context
[Add any additional context, screenshots, or information that may be relevant to the issue.]

---

### Related Code

`   
    
    processed_sample_values = [5 , 10 ]  
    latent_dim_values = [2 , 4 , 8 ]  
    size_values = [7 , 8 , 9  ]  
    epochs = [ 10 , 20 ]
    results = {}
    pd_results = []
    
    SAVE=True
    for processed_sample in processed_sample_values:
        for latent_dim in latent_dim_values:
            for size in size_values:
                for epoch in epochs:
                    fout = open("summary_detailed_5.txt", "a")
                    for i in range(3):
                        model_name = 'dina_model_sample_%d_dim_%d_size_%d_epochs_%d_index_%d.keras' % (processed_sample, latent_dim, size, epoch , i )
                        print(model_name)
                        #fout.write(model_path+"\n")
                        print(f"Running with latent_dim={latent_dim}, epoch={epoch} , processed_sample_num={processed_sample} , matrix_size = {size} x {size}")
                        collected_data = data_autoencoding.main(latent_dim, model_name, processed_sample , size , SAVE , epoch )
                        #creating txt 
                        fout.write(str(collected_data)+"\n"+"\n")
                        #creating pandas dataframe for heatmap
                        pd_results.append({
                            "Model": model_name,
                            "Latent Dimension": latent_dim,
                            "Matrix Size": size,
                            "Processed Sample": processed_sample,
                            "Epochs": epoch,
                            "Spearman Correlation": collected_data["spearman_correlation"],
                            "Spearman p-value": collected_data["spearman_p_value"],
                            "Spearman Correlation Dim1_2": collected_data["spearman_correlation_dim1_2"],
                            "Spearman p-value Dim1_2": collected_data["spearman_p_value_dim1_2"]
                        })
    fout.close()
    df = pd.DataFrame(pd_results)
    df.to_csv("results_2.csv", index=False)

    return df

`
---

### Priority
[Set the priority level for this issue (e.g., high, medium, low).]

---

### Due Date
[Specify a due date, if necessary.]

---
### Checklist for coding 
- [x] Modul configuration.
- [x] Run on several senerios. 
- [x] Saved the run models  .
- [x] Plotting .

### Checklist for tasks 
- [x] Reviewed the problem statement and proposed solution.
- [x] Environmental dependecy solutions.
- [x] Provided expected and actual behavior matches.
- [x] Included all relevant context and code snippets.
- [x] Commited.
